src/test/java/com/es/phoneshop/web/ProductListPageServletTest.java

```diff
@@ -37,5 +44,9 @@ public void testDoGet() throws ServletException, IOException {
         servlet.doGet(request, response);
 
         verify(requestDispatcher).forward(request, response);
+
+        ArgumentCaptor<List> argumentCaptor = ArgumentCaptor.forClass(List.class);
+        verify(request).setAttribute(eq("products"), argumentCaptor.capture());
+        assertEquals(argumentCaptor.getValue(), productDao.findProducts());
```

Comment:
Does productDao.findProducts() return null here? So you checking argumentCaptor.getValue() == null in fact? I guess you forgot to mock return from productDao.findProducts()

Reply:
productDao.findProducts() returns collection with zero size, so I checking this condition

Reply:
Ok, that's true. But the problem is that it's **new** collection object, which is different every time you call this method (even though it's always empty - guess it's default mock that Mockito provides). 
The idea was to check that "product" attribute contains exactly same collection object that was returned from `productDao.findProducts()` call inside `servlet.doGet`